unittest/unit/update-bcd.js

```diff
@@ -14,13 +14,330 @@
+    it('equal', () => {
+      assert.equal(isEquivalent({a: 1, b: 2}, {a: 1, b: 2}), true);
+    });
+
+    it('not equal: different keys', () => {
+      assert.equal(isEquivalent({a: 1, b: 2}, {a: 1, b: 2, c: 3}), false);
+    });
+
+    it('not equal: same keys, different values', () => {
+      assert.equal(isEquivalent({a: 1, b: 2}, {a: 1, b: 3}), false);
+    });
+  });
+
+  describe('findEntry', () => {
+    it('equal', () => {
+      assert.deepEqual(
+          findEntry(bcd, 'api.AbortController'), bcd.api.AbortController
+      );
+    });
+
+    it('no path', () => {
+      assert.equal(findEntry(bcd, ''), null);
+    });
+
+    it('invalid path', () => {
+      assert.equal(findEntry(bcd, 'api.MissingAPI'), undefined);
+    });
+  });
+
+  describe('getBrowserAndVersion', () => {
+    it('Chrome', () => {
+      assert.deepEqual(getBrowserAndVersion('Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/85.0.4183.121 Safari/537.36', bcd.browsers), ['chrome', '85']);
+    });
+
+    it('Chrome Android', () => {
+      assert.deepEqual(getBrowserAndVersion('Mozilla/5.0 (Linux; Android 11; Pixel 2) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/85.0.4183.101 Mobile Safari/537.36', bcd.browsers), ['chrome_android', '85']);
+    });
+
+    it('Edge (EdgeHTML)', () => {
+      assert.deepEqual(getBrowserAndVersion('Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/58.0.3029.110 Safari/537.36 Edge/16.16299', bcd.browsers), ['edge', '16']);
+    });
+
+    it('Safari', () => {
+      assert.deepEqual(getBrowserAndVersion('Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_6) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/14.0 Safari/605.1.15', bcd.browsers), ['safari', '14']);
+    });
+
+    it('Safari iOS', () => {
+      assert.deepEqual(getBrowserAndVersion('Mozilla/5.0 (iPhone; CPU iPhone OS 14_0_1 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/14.0 Mobile/15E148 Safari/604.1', bcd.browsers), ['safari_ios', '14']);
+    });
+
+    it('Samsung Internet (12.0)', () => {
+      assert.deepEqual(getBrowserAndVersion('Mozilla/5.0 (Linux; Android 11; Pixel 2) AppleWebKit/537.36 (KHTML, like Gecko) SamsungBrowser/12.0 Chrome/79.0.3945.136 Mobile Safari/537.36', bcd.browsers), ['samsunginternet_android', '12.0']);
+    });
+
+    it('Samsung Internet (12.1)', () => {
+      assert.deepEqual(getBrowserAndVersion('Mozilla/5.0 (Linux; Android 11; Pixel 2) AppleWebKit/537.36 (KHTML, like Gecko) SamsungBrowser/12.1 Chrome/79.0.3945.136 Mobile Safari/537.36', bcd.browsers), ['samsunginternet_android', '12.1']);
+    });
+
+    it('Yandex Browser (not in BCD)', () => {
+      assert.deepEqual(getBrowserAndVersion('Mozilla/5.0 (Windows NT 6.3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/58.0.3029.110 YaBrowser/17.6.1.749 Yowser/2.5 Safari/537.36', bcd.browsers), [null, null]);
+    });
+  });
+
+  describe('getSupportMap', () => {
```

Comment:
Can you also test `getSupportMatrix`? That's fairly important in terms of managing conflicting results for the same version, and seems to not actually do it in a deliberate way, instead the last report wins, it seems.

Reply:
Of course, adding those tests right now actually!  (This PR's still in drafting stages.)